### ISSUE TITLE
Clang-cl fails at cubeb_wasapi with LNK2019

### ISSUE THREAD

**Issue by MrAlex94** [kinetiknz/cubeb#55]
I realise this may not be any priority at all, but thought it might help raising an issue either way! I'll try to resolve this by myself as well. 
Here's the error:

> cubeb_wasapi.obj : error LNK2019: unresolved external symbol _InterlockedCompareExchange64 referenced in function "int __cdecl `anonymous namespace'::wasapi_stream_get_position(struct cubeb_stream *,unsigned __int64 *)" (?wasapi_stream_get_position@?A@@YAHPEAUcubeb_stream@@PEA_K@Z)

I've tried including winbase.h and even defining InterlockedExchangeAdd64, but still doesn't seem to do any good.

Forgot to mention, trying to build with Clang 3.7.0 64-Bit.

**Comment by kinetiknz** [comment]
It should be possible to use InterlockedCompareExchange64 with clang-cl, but I don't know what magic invocation is required.  If you find it, please do let me know.

In the mean time, this (untested) patch should get the build working: https://gist.github.com/kinetiknz/8e83547e75171b2abd65

**Comment by MrAlex94** [comment]
I wonder what clang-cl is doing, because it still spits out the same error with the patch. Does that mean that ifndef InterlockedExchangeAdd64 isn't being triggered because it *is* being defined but for some reason the linker can't resolve it?

I wonder if it's worth opening a bug report on LLVM bugzilla.

**Comment by kinetiknz** [comment]
Had a chance to set this build environment up locally and poke around.  My original patch doesn't work because InterlockedExchangeAdd64 is #defined as _InterlockedExchangeAdd64, so the #ifndef path is never built.  _InterlockedExchangeAdd64 is an inline in winnt.h that uses InterlockedCompareExchange64 which is #defined as _InterlockedCompareExchange64.  For some reason I haven't tracked down, the intrinsic backing that isn't available until you #include intrin.h.

The short version is: this can be fixed by just including intrin.h.

**Comment by kinetiknz** [comment]
Fixed by a3091e55bdfdd3358dd04840b36a0d7568acbfab.
